src/java.base/share/classes/java/util/concurrent/LinkedBlockingDeque.java

```diff
@@ -579,7 +581,7 @@ public E peekLast() {
         }
     }
 
-    public boolean removeFirstOccurrence(Object o) {
+    public boolean removeFirstOccurrence(@Nullable Object o) {
```

Comment:
Should this change (and the ones to `removeLastOccurrence` and `remove` below) go into [collection-object-parameters-may-be-null.astub](https://github.com/eisop/checker-framework/blob/master/checker/src/main/java/org/checkerframework/checker/nullness/collection-object-parameters-may-be-null.astub) instead?
I see some entries there, e.g. for `Deque`.

Reply:
My impression has been that these files haven't quite reached a consistent state.

`LinkedBlockingDeque` already has `@Nullable` on its `contains` method. That method's docs don't mention `null` explicitly, but they do say...

> this deque contains at least one element `e` such that `o.equals(e)`

...which at best implies that `null` is an acceptable input.

The methods I'm annotating in this PR are similar. So I think it would be defensible to remove the annotation on `contains`, rather than add the ones in this PR. (And as you say, we could update [collection-object-parameters-may-be-null.astub](https://github.com/eisop/checker-framework/blob/master/checker/src/main/java/org/checkerframework/checker/nullness/collection-object-parameters-may-be-null.astub) to have the annotations for all the methods.)

I can do that, but I'll wait for confirmation just in case. But probably nothing I've said above will come as a surprise :)

Reply:
It unfortunately looks like there is some inconsistency here about which methods take `@Nullable Object` here or in `collection-object-parameters-may-be-null.astub` and I don't see a pattern.
As most of the changes in this PR are related to this, I think it would be great to merge the other changes and separately clean these methods.